ntoskrnl/mm/section.c

```diff
@@ -4871,6 +4874,9 @@ MmPurgeSegment(
         PurgeEnd.QuadPart = PageTable->FileOffset.QuadPart + _countof(PageTable->PageEntries) * PAGE_SIZE;
     }
 
+    /* Find byte offset of the page to start */
+    PurgeStart.QuadPart = PAGE_ROUND_DOWN(PurgeStart.QuadPart);
```

Comment:
Could you move this up to where PurgeStart gets initialized? (See also the other suggestions)

Reply:
This code is not needed when there's nothing to find/flush/purge. Moving it up can also leads to a confusion when doing step-by-step debugging.